csharp/test/Microsoft.ML.OnnxRuntime.Tests/InferenceTest.cs

```diff
@@ -2205,6 +2205,30 @@ private void TestSharedAllocatorUsingCreateAndRegisterAllocator()
             }
         }
 
+        [Fact]
+        private void TestCUDAProviderOptions()
+        {
+            #if USE_CUDA
+            using (var cudaProviderOptions = new OrtCUDAProviderOptions())
+            {
+                var providerOptionsDict = new Dictionary<string, string>();
+                providerOptionsDict["device_id"] = "0";
+                providerOptionsDict["arena_extend_strategy"] = "kSameAsRequested";
+                providerOptionsDict["cuda_mem_limit"] = "200000";
+                providerOptionsDict["cudnn_conv_algo_search"] = "HEURISTIC";
+                providerOptionsDict["do_copy_in_default_stream"] = "1";
+
+                cudaProviderOptions.UpdateOptions(providerOptionsDict);
```

Comment:
>cudaProviderOptions.UpdateOptions(providerOptionsDict); [](http://example.com/codeflow?start=16&length=55)

Is there any way to make sure that they actually had effect? Also can we add some negative test?